win32ss/gdi/ntgdi/freetype.c

```diff
@@ -6228,118 +6237,31 @@ IntExtTextOutW(
-               IntUpdateBoundsRect(dc, &DestRect);
-            }
-            IntEngBitBlt(
-                &psurf->SurfObj,
-                NULL,
-                NULL,
-                (CLIPOBJ *)&dc->co,
-                NULL,
-                &DestRect,
-                &SourcePoint,
-                &SourcePoint,
-                &dc->eboBackground.BrushObject,
-                &BrushOrigin,
-                ROP4_FROM_INDEX(R3_OPINDEX_PATCOPY));
-            MouseSafetyOnDrawEnd(dc->ppdev);
-            BackgroundLeft = DestRect.right;
-
-            DestRect.left = ((TextLeft + 32) >> 6) + realglyph->left;
-            DestRect.right = DestRect.left + realglyph->bitmap.width;
-            DestRect.top = TextTop + yoff - realglyph->top;
-            DestRect.bottom = DestRect.top + realglyph->bitmap.rows;
-
-            bitSize.cx = realglyph->bitmap.width;
-            bitSize.cy = realglyph->bitmap.rows;
-            MaskRect.right = realglyph->bitmap.width;
-            MaskRect.bottom = realglyph->bitmap.rows;
-
-            if (NULL == Dx)
-            {
-                TextLeft += realglyph->root.advance.x >> 10;
-                DPRINT("New TextLeft: %I64d\n", TextLeft);
-            }
-            else
-            {
-                // FIXME this should probably be a matrix transform with TextTop as well.
-                Scale = pdcattr->mxWorldToDevice.efM11;
-                if (FLOATOBJ_Equal0(&Scale))
-                    FLOATOBJ_Set1(&Scale);
-
-                /* do the shift before multiplying to preserve precision */
-                FLOATOBJ_MulLong(&Scale, Dx[i<<DxShift] << 6);
-                TextLeft += FLOATOBJ_GetLong(&Scale);
-                DPRINT("New TextLeft2: %I64d\n", TextLeft);
-            }
-
-            if (DxShift)
-            {
-                TextTop -= Dx[2 * i + 1] << 6;
-            }
-
-            previous = glyph_index;
-
-            if (EmuBold || EmuItalic)
-            {
-                FT_Done_Glyph((FT_Glyph)realglyph);
-                realglyph = NULL;
-            }
-        }
+        DestRect.left = (RealXStart + 32) >> 6;
+        DestRect.right = (RealXStart + TextWidth + 32) >> 6;
+        DestRect.top = YStart + yoff - ((fixAscender + 32) >> 6);
+        DestRect.bottom = DestRect.top + ((fixAscender + fixDescender) >> 6);
+
+        MouseSafetyOnDrawStart(dc->ppdev, DestRect.left, DestRect.top, DestRect.right, DestRect.bottom);
+        if (dc->fs & (DC_ACCUM_APP | DC_ACCUM_WMGR))
+        {
+            IntUpdateBoundsRect(dc, &DestRect);
+        }
+        IntEngBitBlt(&psurf->SurfObj,
+                     NULL,
+                     NULL,
+                     (CLIPOBJ *)&dc->co,
+                     NULL,
+                     &DestRect,
+                     &SourcePoint,
+                     &SourcePoint,
+                     &dc->eboBackground.BrushObject,
+                     &BrushOrigin,
+                     ROP4_FROM_INDEX(R3_OPINDEX_PATCOPY));
```

Comment:
![VirtualBox_ReactOS_09_11_2022_10_54_40](https://user-images.githubusercontent.com/2107452/200721650-82b63a5b-ad5a-424e-89f3-9aa02ac16e2f.png)
Why is the background not filled correctly?

Reply:
Was it filled correctly without the PR?

Reply:
Step 1:
![1](https://user-images.githubusercontent.com/2107452/200724066-6ccbe706-75ad-425c-b511-7f8731637e0a.png)

Step 2:
![2](https://user-images.githubusercontent.com/2107452/200724068-30c50671-1180-43c7-bce2-168c74cec959.png)

Step 3:
![3](https://user-images.githubusercontent.com/2107452/200724351-0950e004-6b60-4746-ae50-98f7de0b8152.png)

Step 4:
![4](https://user-images.githubusercontent.com/2107452/200724343-b19b09e0-d0f8-4a9e-949c-6871c218c4be.png)

Step 5:
![5](https://user-images.githubusercontent.com/2107452/200724059-70506798-e1a2-417e-bcf5-2bac8695f5d7.png)

Refreshment is something wrong...

Reply:
> Was it filled correctly without the PR?

Before this PR, the background is filled in main loop.